dhis-2/dhis-support/dhis-support-hibernate/src/main/java/org/hisp/dhis/hibernate/DefaultHibernateConfigurationProvider.java

```diff
@@ -232,6 +232,22 @@ private Properties getHibernateProperties()
         putIfExists( configurationProvider.getProperty( ConfigurationKey.CONNECTION_USERNAME ), Environment.USER, props );
         putIfExists( configurationProvider.getProperty( ConfigurationKey.CONNECTION_PASSWORD ), Environment.PASS, props );
         putIfExists( configurationProvider.getProperty( ConfigurationKey.CONNECTION_POOL_MAX_SIZE ), Environment.C3P0_MAX_SIZE, props );
+        putIfExists( configurationProvider.getProperty( ConfigurationKey.CONNECTION_POOL_MIN_SIZE ),
+            ConfigurationKey.CONNECTION_POOL_MIN_SIZE.getKey(), props );
```

Comment:
Here we need to use the Hibernate Environment equivalent @netroms as the second argument . E.g. use `Environment.C3P0_MAX_SIZE`. This because we are eventually passing these properties to Hibernate. Hibernate needs its own config properties, not the DHIS 2 specific properties.

Reply:
Or with the Java config change in `HibernateConfig` , maybe this is no longer needed, and we could refactor and not use the Hibernate property names at all?

Reply:
Hi, yes I have tested it, the properties are propagated as normal properties through the Configuration object

Reply:
That's good. I think it's time for a refactor of the config stuff but that can certainly wait.